blueflood-http/src/main/java/com/rackspacecloud/blueflood/http/MediaTypeChecker.java

```diff
@@ -0,0 +1,46 @@
+package com.rackspacecloud.blueflood.http;
+
+import com.google.common.base.Strings;
+import io.netty.handler.codec.http.*;
+
+/**
+ * A small utility class to be used by our Ingest and Query handlers to check for
+ * valid media-type related headers.
+ */
+public class MediaTypeChecker {
+
+    private static final String MEDIA_TYPE_APPLICATION_JSON = "application/json";
+    private static final String ACCEPT_ALL = "*/*";
+
+    /**
+     * Checks the Content-Type header to see if clients specify the right
+     * media type
+     * @param headers
+     * @return
+     */
+    public boolean isContentTypeValid(HttpHeaders headers) {
+
+        String contentType = headers.get(HttpHeaders.Names.CONTENT_TYPE);
+
+        // if we get no Content-Type or we get application/json, then it's valid
+        // any other, it's invalid
+        return (Strings.isNullOrEmpty(contentType) ||
+                contentType.contains(MEDIA_TYPE_APPLICATION_JSON));
+    }
+
+    /**
+     * Checks the Accept header to see if it clients accept the correct
```

Comment:
I think it would make sense of "it" were simply replaced with "the". That's probably what she meant.